### PR TITLE
Fix ordering of testcase declarations in src/app/tests/

### PR DESCRIPTION
**Pull request by mbknust**
Previously, the order in which testcase functions were defined, was different than the order in which they are supposed to execute.
This inconsistency matters, if we want to move away from the deprecated `nlunit-test` framework. 
This PR does not add or remove any code, it only changes the order in which those functions are defined.

### CODE REVIEW

**Review by arkq — APPROVED** [review]

**Comment by andy31415** [comment]
@mbknust are these tests order-dependent? If yes that sounds concerning. If no, would a change of order in the list of tests to run result in a smaller to review PR? Actually checking if these 2.5 k lines were just moved or any errors happened would be very hard and we also lose a lot of `git blame` history when moving lines (so we do not know when something was actually changed or not)

**Comment by mbknust** [comment]
> @mbknust are these tests order-dependent? If yes that sounds concerning. If no, would a change of order in the list of tests to run result in a smaller to review PR? 

Yes, unfortunately they are order dependent. If the tests are run in the wrong order, some assertions will fail.

> Actually checking if these 2.5 k lines were just moved or any errors happened would be very hard

That's true, though at least I was able to do a quick sanity check: sorting the lines of both files (before and after the commit) gives the same result

**Comment by arkq** [comment]
@andy31415 @mbknust Fixing tests order dependency would be preferable, but I'm not sure how easy that would be. There are other Test* files where order is important. I do not remember which test file that was, but I was trying to fix some issues, and in order to narrow the scope and speedup development I commented out all tests form the beginning to the affected tests case, and the result was test failure on my branch and master (to my surprise). Tracking test order dependency is easy with gtest which have an option to randomize tests execution... so after moving to pigweed test it would be easier (if pigweed test supports such feature).

Here, we can minimize the number of lines moved around by checking which exactly tests need to be in order and move only these, and for the rest of them change test order in the test list.

File: src/app/tests/TestCommandInteraction.cpp [hunk 1]
```diff
@@ -475,242 +475,6 @@ void TestCommandInteraction::AddInvokeResponseData(nlTestSuite * apSuite, void *
     }
 }
 
-void TestCommandInteraction::TestCommandSenderWithWrongState(nlTestSuite * apSuite, void * apContext)
```
Comment (arkq) [inline src/app/tests/TestCommandInteraction.cpp:478]: Adding blocking comment to check which tests actually need to be in order and move only these (to reduce the number of changed lines).

**Review by arkq — CHANGES_REQUESTED** [review]

File: src/app/tests/TestCommandInteraction.cpp (hunk 1 above)
Comment (bzbarsky-apple) [inline src/app/tests/TestCommandInteraction.cpp:478]: There should be no ordering dependencies between tests here... If there are, that needs to be fixed.

**Review by bzbarsky-apple — CHANGES_REQUESTED** [review]

**Comment by arkq** [comment]
@mbknust Please check whether reorder in the `TestCommandInteraction.cpp` is indeed required. I've checked running all these tests cases one by one and none failed ...but maybe the dependency between these tests is more tricky, or something has changed in master since your initial work on pigweed test adaptation.

**Comment by bzbarsky-apple** [comment]
@tehampson you were in that code recently... any idea offhand why those tests might be stateful?

**Pull request closed without merging**